src/main/java/LinkedStack.java

```diff
@@ -100,16 +106,10 @@ public int hashCode() {
     }
 
     /**
-     * A Node class for a singly linked structure. Each node
-     * contains a reference to data of type T, which may be
-     * null, and a reference to the next/subsequent/successor
-     * singly linked node, which may also be null.
-     * <p>
-     * This class is a static nested class since the node
-     * class is only needed for the implementation of the
-     * LinkedStack.
+     * A Node class for a singly linked structure. Each node contains a reference to data of type T, which may be null,
```

Comment:
`T` can't be `null` -- null isn't a type in java. More like _Each node contains a nullable reference to data of type `T`_.